File: LEC15/LEC15_01.py
a = input()
b = input()
a = a.lower()
b = b.lower()
max_length = max(len(a), len(b))
dp = [[0 for _ in range(len(b)+1)] for _ in range(len(a)+1)]

for i in range(0, len(a)+1):
    for j in range(0, len(b)+1):
        if i == 0:
            dp[i][j] = j
        elif j == 0:
            dp[i][j] = i
        elif a[i-1] == b[j-1]:
            dp[i][j] = dp[i-1][j-1]
        else:
            dp[i][j] = min(
                dp[i-1][j],
                dp[i][j-1],
                dp[i-1][j-1]
            ) + 1

edit = dp[len(a)][len(b)]
isSub = (max(len(a), len(b))+1) // 2

if edit < isSub:
    print(edit, 1)
else:
    print(edit, 0)

File: LEC15/LEC15_01.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
  string a, b;
  cin >> a >> b;
  transform(a.begin(), a.end(), a.begin(), ::tolower);
  transform(b.begin(), b.end(), b.begin(), ::tolower);
  int max_length = max(a.length(), b.length());
  int dp[a.length() + 1][b.length() + 1];
  for (int i = 0; i <= a.length(); i++)
  {
    for (int j = 0; j <= b.length(); j++)
    {
      if (i == 0)
      {
        dp[i][j] = j;
      }
      else if (j == 0)
      {
        dp[i][j] = i;
      }
      else if (a[i - 1] == b[j - 1])
      {
        dp[i][j] = dp[i - 1][j - 1];
      }
      else
      {
        dp[i][j] = min(
                       dp[i - 1][j],
                       min(dp[i][j - 1],
                           dp[i - 1][j - 1])) +
                   1;
      }
    }
  }
  int edit = dp[a.length()][b.length()];
  int isSub = (max(a.length(), b.length()) + 1) / 2;
  if (edit < isSub)
  {
    cout << edit << " " << 1 << endl;
  }
  else
  {
    cout << edit << " " << 0 << endl;
  }
  return 0;
}